doc/authoring_command_modules/README.md

```diff
@@ -75,63 +75,77 @@ Command modules should have the following structure:
 |           `-- <MODULE_NAME>
 |               `-- __init__.py
 `-- setup.py
+`-- HISTORY.rst
 ```
 
 **Create an \_\_init__.py for your module**
 
-In the \_\_init__ file, two methods need to be defined:
-  - `load_commands` - Uses the file in the 'Writing a Command' section below to load the commands.
-  - `load_params` - Uses the file in the 'Customizing Arguments' section below to load parameter customizations.
+In the \_\_init__ file, you will declare a command loader class that inherits from AzCommandsLoader. You will typically override the following three methods:
+  - `__init__` - Useful for setting metadata that applies to the entire module. See command authoring for more info.
+  - `load_commands_table` - Register command groups and commands here. It is common to store the implementation of this method in
+                            a file named `commands.py` but for very small modules this may not be necessary. See command authoring for
+                            more info.
+  - `load_arguments` - Apply metadata to your command arguments. It is common to store the implementation of this method in a file 
+                       named `_params.py` but for very small modules this may not be necessary. See command authoring for more info.
 
+**__init__.py**
 ```Python
-def load_params(command):
-    import azure.cli.command_modules.<module_name>._params
+from azure.cli.core import AzCommandsLoader
+from azure.cli.command_modules.example._help import helps  # pylint: disable=unused-import
 
-def load_commands():
-    import azure.cli.command_modules.<module_name>.commands
+class ExampleCommandsLoader(AzCommandsLoader):
+
+    def load_command_table(self, args):
+      from azure.cli.core.commands import CliCommandType
+
+      example_custom = CliCommandType(
+        operations_tmpl='azure.mgmt.example.operations#ExampleOperations.{}',
+      )
+
+      with self.command_group('example', example_custom) as g:
+        g.command('create', 'create_example')
+
+COMMAND_LOADER_CLS = ExampleCommandsLoader
 ```
 
+**custom.py**
 ```python
-from azure.cli.core.commands import cli_command
+def create_example(cmd, example_name, resource_group_name, location=None):
```

Comment:
Nitpick - I would use a different name of the command, because this can be interpreted as it actually creating an example. And we do expect commands to have, well, examples :).

Reply:
Fair enough :)